### PR TITLE
[#544] Fix removing grafana dashboard

### PR DESCRIPTION
**Pull request by skayerok**
Fixed removing whole grafana dashboard when undeploying one version of multiversion model

### CODE REVIEW

**Comment by skayerok** [comment]
closes #544

**Comment by skayerok** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

File: tests/robot/tests/4_edi_multi_versioned_models/4.0_check_edi_multi_versioned_models.robot [hunk 1]
```diff
@@ -36,8 +37,12 @@ Check EDI undeploy 1 of 2 models with different versions but the same id
                     Should not contain              ${resp.stdout}          ${TEST_MODEL_IMAGE_1}
                     Should contain                  ${resp.stdout}          ${TEST_MODEL_IMAGE_2}
 
+                    Connect to enclave Grafana      ${MODEL_TEST_ENCLAVE}
+
+                    Dashboard should exists         ${TEST_MODEL_ID}
+
 Check EDI undeploy all model instances by version
-    [Tags]  edi  cli  enclave  multi_versions  apps
+    [Tags]  edi  cli  enclave  multi_versions  apps  unique
```
Comment (mcdoker18) [inline tests/robot/tests/4_edi_multi_versioned_models/4.0_check_edi_multi_versioned_models.robot]: Remove tag

File: tests/robot/tests/4_edi_multi_versioned_models/4.0_check_edi_multi_versioned_models.robot [hunk 2]
```diff
@@ -55,6 +60,12 @@ Check EDI undeploy all model instances by version
                     Should not contain              ${resp.stdout}          |${TEST_MODEL_1_VERSION}
                     Should contain                  ${resp.stdout}          |${TEST_MODEL_2_VERSION}
 
+                    Connect to enclave Grafana      ${MODEL_TEST_ENCLAVE}
+                    Dashboard should exists         ${TEST_MODEL_ID}
+    ${resp}=        Run EDI undeploy with version   ${MODEL_TEST_ENCLAVE}   ${TEST_MODEL_ID}    ${TEST_MODEL_2_VERSION}
+                    Dashboard should not exists     ${TEST_MODEL_ID}
+                    Should Be Equal As Integers     ${resp.rc}              0
```
Comment (mcdoker18) [inline tests/robot/tests/4_edi_multi_versioned_models/4.0_check_edi_multi_versioned_models.robot]: Change the command order
```robot
Should Be Equal As Integers     ${resp.rc}              0
Dashboard should not exists     ${TEST_MODEL_ID}
```

File: k8s/edge/Dockerfile [hunk 3]
```diff
@@ -13,21 +13,12 @@
...
-  && apk add --no-cache --virtual .install-deps openssl libffi-dev openssl-dev \
-       python3 python3-dev bash g++ zlib zlib-dev jpeg jpeg-dev \
-       ca-certificates gnupg openssl git curl \
-  && wget "https://github.com/Yelp/dumb-init/releases/download/v${DUMB_INIT_VERSION}/dumb-init_${DUMB_INIT_VERSION}_amd64" \
-  && wget "https://github.com/Yelp/dumb-init/releases/download/v${DUMB_INIT_VERSION}/sha256sums" \
-  && grep "dumb-init_${DUMB_INIT_VERSION}_amd64$" sha256sums | sha256sum -c \
-  && rm sha256sums \
-  && mv dumb-init_${DUMB_INIT_VERSION}_amd64 /usr/bin/dumb-init \
-  && chmod +x /usr/bin/dumb-init
+RUN apt update -y && \
```
Comment (mcdoker18) [inline k8s/edge/Dockerfile]: Please, get rid of the my commit

File: legion/legion/edi/server.py [hunk 4]
```diff
@@ -208,9 +208,11 @@ def undeploy(model, version=None, grace_period=0, ignore_not_found=False):
...
         if app.config['REGISTER_ON_GRAFANA']:
+            other_versions_exist = len(app.config['ENCLAVE'].get_models(model_service.id)) > 1
             LOGGER.info('Removing model\'s dashboard from Grafana (id={}, version={})'
                         .format(model_service.id, model_service.version))
-            if app.config['GRAFANA_CLIENT'].is_dashboard_exists(model_service.id, model_service.version):
+            if app.config['GRAFANA_CLIENT'].is_dashboard_exists(model_service.id, model_service.version) \
+                    and not other_versions_exist:
                 app.config['GRAFANA_CLIENT'].remove_dashboard_for_model(model_service.id,
                                                                         model_service.version)
             else:
```
Comment (mcdoker18) [inline legion/legion/edi/server.py]:
```python
LOGGER.info('Removing model\'s dashboard from Grafana has been skipped - \
                              dashboard does not exist')
```
The log is not entirely correct, what do you think about the following version?
```python
LOGGER.info('Removing model\'s dashboard from Grafana has been skipped')
```

Comment (skayerok) [reply legion/legion/edi/server.py]: Agreed, this looks more accurate

**Pull request merged**